### PR TITLE
[Tag] fill & [TagInput] intent props

### PR DESCRIPTION
**Pull request by giladgray**
#### Changes proposed in this pull request:

- add `<Tag fill>` prop to make it a block element.
- add `<TagInput intent>` prop to support colors.

### CODE REVIEW

File: packages/core/src/components/tag/tag.tsx [hunk 1]
```diff
@@ -101,6 +108,7 @@ export class Tag extends React.PureComponent<ITagProps, {}> {
             Classes.intentClass(intent),
             {
                 [Classes.ACTIVE]: active,
+                [Classes.FILL]: fill,
```
Comment (adidahiya) [inline packages/core/src/components/tag/tag.tsx]: there is no corresponding CSS change, so this doesn't seem to do anything

Comment (invliD) [reply packages/core/src/components/tag/tag.tsx]: https://github.com/palantir/blueprint/blob/develop/packages/core/src/common/_flex.scss#L8

Comment (adidahiya) [reply packages/core/src/components/tag/tag.tsx]: are you able to get it working in the docs? I can't

Comment (invliD) [reply packages/core/src/components/tag/tag.tsx]: The parent in the docs isn't a `@pt-flex-container`, so no.

Why does `.pt-flex` only work inside a `@pt-flex-container`? Why not just globally allow it? I doubt consumers use the mixin, I certainly never have.

Comment (giladgray) [reply packages/core/src/components/tag/tag.tsx]: i added this prop to support flex containers, as you've noticed. i'm happy to add some corresponding styles (`display: flex; width: 100%`) if you'd like to see that too @adidahiya @invliD?

Comment (adidahiya) [reply packages/core/src/components/tag/tag.tsx]: yeah, I would like to see some way to enable this style modifier in the docs, thanks

Comment (giladgray) [reply packages/core/src/components/tag/tag.tsx]: done

**Pull request merged**